include/pybind11/detail/internals.h

```diff
@@ -100,7 +100,9 @@ struct internals {
     std::unordered_set<std::pair<const PyObject *, const char *>, overload_hash> inactive_overload_cache;
     type_map<std::vector<bool (*)(PyObject *, void *&)>> direct_conversions;
     std::unordered_map<const PyObject *, std::vector<PyObject *>> patients;
+#if !defined (__PGIC__)
```

Comment:
```suggestion
#ifndef __PGIC__
```

This is maybe a little cleaner for a single check?

Reply:
That is a question of taste.
Later, when the exceptions will be implemented, that should be something like

```
#if !defined (__PGIC__)|| (__PGIC_MAJOR__ > 21)
```

Reply:
Okay, fair enough.